gaeul/source/source-application.c

```diff
@@ -250,6 +254,42 @@ gaeguli_nest_dbus_register (GaeguliNest * nest, GDBusConnection * connection)
   }
 }
 
+static void
+_stream_quality_dropped_cb (GaeguliNest * nest)
+{
+  g_autoptr (GError) error = NULL;
+
+  nest->record_stream =
+      gaeguli_pipeline_add_recording_target (nest->pipeline,
+      nest->record_location, &error);
+  gaeguli_target_start (nest->record_stream, &error);
+}
+
+static void
+_stream_quality_regained_cb (GaeguliNest * nest)
+{
+  g_autoptr (GError) error = NULL;
+
+  gaeguli_pipeline_remove_target (nest->pipeline, nest->record_stream, &error);
+
+  g_debug ("target removed");
+}
+
+static void
+_stream_started_cb (GaeguliPipeline * pipeline, guint target_id,
+    gpointer user_data)
+{
+  GaeguliNest *nest = user_data;
+
+  nest->adaptor = gaeguli_target_get_stream_adaptor (nest->target_stream);
```

Comment:
I am not sure whether the pipeline emits stream-start only once.
We should avoid memory leakage.